src/pages/studyView/StudyViewUtils.tsx

```diff
@@ -981,6 +992,57 @@ export function calculateLayout(visibleAttributes: ChartMeta[], cols: number): L
     return layout;
 }
 
+export function calculateNewLayoutForFocusedChart(previousLayout: Layout, currentFocusedChartByUser: ChartMeta, cols: number): Layout {
+    const initialX = previousLayout.x;
+    const initialY = previousLayout.y;
+    const dimensionWidth = currentFocusedChartByUser.dimension.w;
+    let x = initialX;
+    let y = initialY;
+
+    if (isFocusedChartShrinked({w: previousLayout.w, h: previousLayout.h}, currentFocusedChartByUser.dimension)) {
+        x = initialX + previousLayout.w - dimensionWidth;
+    }
+    else if (initialX + dimensionWidth >= cols && initialX - dimensionWidth >= 0) {
+        x = cols - dimensionWidth;
+    }
+
+    return {
+        i: currentFocusedChartByUser.uniqueKey,
+        x,
+        y,
+        w: currentFocusedChartByUser.dimension.w,
+        h: currentFocusedChartByUser.dimension.h,
+        isResizable: false
+    };
+}
+
+export function populateFixedPositionLayout(layout: Layout, matrix: string[][], cols: number) {
+    const xMax = layout.x + layout.w;
+    const yMax = layout.y + layout.h;
+    while (yMax >= matrix.length) {
+        matrix.push(new Array(cols).fill(''));
```

Comment:
lets not use fill method

Reply:
We may don't need to change this since babel_polyfill handled this function.

Reply:
Just curious why we should not use fill method? @alisman